sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathPermissions.cs

```diff
@@ -0,0 +1,249 @@
+        /// Public constructor.
+        /// </summary>
+        /// <param name="owner">The path owner's permissions.</param>
+        /// <param name="group">The path group's permissions.</param>
+        /// <param name="other">Permissions for other users.</param>
+        /// <param name="stickyBit">If sticky bit is enabled</param>
+        /// <param name="extendedInfoInAcl">If there is extended info in the ACL</param>
+        public PathPermissions(
+            RolePermissions owner,
+            RolePermissions group,
+            RolePermissions other,
+            bool stickyBit = false,
+            bool extendedInfoInAcl = false)
+        {
+            Owner = owner;
+            Group = group;
+            Other = other;
+            StickyBit = stickyBit;
+            ExtendedInfoInAcl = extendedInfoInAcl;
+        }
+
+        /// <summary>
+        /// Parses a string in octal format to PathPermissions.
+        /// </summary>
+        /// <param name="octalString">Octal string to parse.</param>
+        /// <returns><see cref="PathPermissions"/>.</returns>
+        public static PathPermissions ParseOctal(string octalString)
+        {
+            if (octalString == null)
+            {
+                return null;
+            }
+
+            if (octalString.Length != 4)
+            {
+                throw new ArgumentException("octalString must be 4 characters");
+            }
+
+            var pathPermissions = new PathPermissions();
+
+            if (octalString[0] == '0')
+            {
+                pathPermissions.StickyBit = false;
+            }
+            else if (octalString[0] == '1')
+            {
+                pathPermissions.StickyBit = true;
+            }
+            else
+            {
+                throw new ArgumentException("First digit of octalString must be 0 or 1");
+            }
+
+            pathPermissions.Owner = RolePermissionsExtensions.ParseOctal(octalString[1]);
+            pathPermissions.Group = RolePermissionsExtensions.ParseOctal(octalString[2]);
+            pathPermissions.Other = RolePermissionsExtensions.ParseOctal(octalString[3]);
+
+            return pathPermissions;
+        }
+
+        /// <summary>
+        /// Parses a symbolic string to PathPermissions.
+        /// </summary>
+        /// <param name="symbolicString">String to parse</param>
+        /// <returns><see cref="PathPermissions"/></returns>
+        public static PathPermissions ParseSymbolic(string symbolicString)
+        {
+            if (symbolicString == null)
+            {
+                return null;
+            }
+
+            if (symbolicString.Length != 9 && symbolicString.Length != 10)
+            {
+                throw new ArgumentException("symbolicString must be 9 or 10 characters");
+            }
+
+            var pathPermissions = new PathPermissions();
+
+            // Set sticky bit
+            if (symbolicString.ToLower(CultureInfo.InvariantCulture)[8] == 't')
```

Comment:
Use `char.ToLower` instead of allocating a new string.

Reply:
fixed.